src/components/Asset/AssetActions/Compute/index.tsx

```diff
@@ -1,232 +1,286 @@
+        initializedProvider?.datasets?.[0]?.providerFee
+      )
+      if (!datasetPriceAndFees) {
+        setError('Error setting dataset price and fees!')
+        return
+      }
+      setDatasetOrderPriceAndFees(datasetPriceAndFees)
+    }
 
-      algorithmSelectionList = await transformAssetToAssetSelection(
-        computeService?.serviceEndpoint,
-        gueryResults.results,
-        []
+    if (
+      selectedAlgorithmAsset?.accessDetails?.addressOrId !== ZERO_ADDRESS &&
+      selectedAlgorithmAsset?.accessDetails?.type !== 'free' &&
+      initializedProvider?.algorithm?.providerFee
+    ) {
+      setComputeStatusText(
+        getComputeFeedback(
+          selectedAlgorithmAsset?.accessDetails?.baseToken?.symbol,
+          selectedAlgorithmAsset?.accessDetails?.datatoken?.symbol,
+          selectedAlgorithmAsset?.metadata?.type
+        )[0]
       )
+      let algoPoolParams = null
+      if (selectedAlgorithmAsset?.accessDetails?.type === 'dynamic') {
+        const response = await getPoolData(
+          selectedAlgorithmAsset.chainId,
+          selectedAlgorithmAsset.accessDetails.addressOrId,
+          selectedAlgorithmAsset?.nft.owner,
+          accountId || ''
+        )
+        algoPoolParams = {
+          tokenInLiquidity: response?.poolData?.baseTokenLiquidity,
+          tokenOutLiquidity: response?.poolData?.datatokenLiquidity,
+          tokenOutAmount: '1',
+          opcFee: getOpcFeeForToken(
+            selectedAlgorithmAsset?.accessDetails?.baseToken.address,
+            selectedAlgorithmAsset?.chainId
+          ),
+          lpSwapFee: response?.poolData?.liquidityProviderSwapFee,
+          publishMarketSwapFee:
+            selectedAlgorithmAsset?.accessDetails?.publisherMarketOrderFee,
+          consumeMarketSwapFee: '0'
+        }
+      }
+      const algorithmOrderPriceAndFees = await getOrderPriceAndFees(
+        web3,
+        selectedAlgorithmAsset,
+        ZERO_ADDRESS,
+        algoPoolParams,
+        initializedProvider.algorithm.providerFee
+      )
+      if (!algorithmOrderPriceAndFees) {
+        setError('Error setting algorithm price and fees!')
+        return
+      }
+      setAlgoOrderPriceAndFees(algorithmOrderPriceAndFees)
     }
-    return algorithmSelectionList
   }
 
-  const initMetadata = useCallback(async (ddo: Asset): Promise<void> => {
-    if (!ddo) return
-    const accessDetails = await getAccessDetails(
-      ddo.chainId,
-      ddo.services[0].datatokenAddress
-    )
-    setAlgorithmConsumeDetails(accessDetails)
-  }, [])
-
   useEffect(() => {
-    if (!algorithmConsumeDetails) return
+    if (!asset?.accessDetails || !accountId) return
 
-    setIsAlgoConsumablePrice(algorithmConsumeDetails.isPurchasable)
-  }, [algorithmConsumeDetails])
+    setIsConsumablePrice(asset?.accessDetails?.isPurchasable)
+    // setIsOwned(asset?.accessDetails?.isOwned)
+    setValidOrderTx(asset?.accessDetails?.validOrderTx)
+  }, [asset?.accessDetails])
```

Comment:
this should get `accountId` as dependency. Unless we are sure if user connects wallet, or switches accounts, this effect actually changes